test/tools/utils.tests.js

```diff
@@ -439,3 +439,98 @@ describe('#utils processChangedObjectFields', () => {
     expect(desiredObjectFieldState).to.deep.equal(desiredAssetState);
   });
 });
+
+describe('#keywordReplacement', () => {
+  it('should replace string keywords and array keywords in a JSON file', () => {
+    const mapping = {
+      ARRAY_REPLACEMENT: [ 'foo', 'bar' ],
+      STRING_REPLACEMENT: 'baz',
+      OTHER_REPLACEMENT: 'lol'
+    };
+    const inputJSON = '{ "arrayReplaceNoQuotes": @@ARRAY_REPLACEMENT@@, "arrayReplaceWithQuotes": "@@ARRAY_REPLACEMENT@@", "stringReplace": "##STRING_REPLACEMENT##", "noReplace": "OTHER_REPLACEMENT" }';
+    const output = utils.keywordReplace(inputJSON, mapping);
+
+    expect(() => JSON.parse(output)).to.not.throw();
+
+    expect(output).to.equal(`{ "arrayReplaceNoQuotes": ${JSON.stringify(mapping.ARRAY_REPLACEMENT)}, "arrayReplaceWithQuotes": ${JSON.stringify(mapping.ARRAY_REPLACEMENT)}, "stringReplace": "${mapping.STRING_REPLACEMENT}", "noReplace": "OTHER_REPLACEMENT" }`);
+  });
+
+  it('should replace keywords in YAML file', () => {
+    const mapping = {
+      ARRAY_REPLACEMENT: [ 'foo', 'bar' ],
+      STRING_REPLACEMENT: 'baz',
+      OTHER_REPLACEMENT: 'lol'
+    };
+
+    const inputYAML = `
+    ---
+    stringReplaceNoQuotes: ##STRING_REPLACEMENT##
+    stringReplaceWithQuotes: "##STRING_REPLACEMENT##"
+    arrayReplace: @@ARRAY_REPLACEMENT@@
+    arrayReplaceWithQuotes: "@@ARRAY_REPLACEMENT@@"
```

Comment:
Unlikely that someone would wrap this value in quotes when using YAML, but at least it's technically supported. I don't believe this behavior will interfere with any current customer workflows.